// grpc_gui.proto
// rocnnet
//
// purpose:
// define service where rocnnet app acts as client

syntax = "proto3";

package visor;

message Empty {}

message ClientId {
    string id = 1;
}

message NodeMessage {
    enum NodeStatus {
        REMOVE = 0;
        UPDATE = 1;
    }

    string id = 1;
    NodeStatus status = 2;
}

message EdgeMessage {
    enum EdgeStatus {
        ADD = 0;
        REMOVE = 1;
    }

    string obsId = 1;
    string subId = 2;
    uint64 idx = 3;
    EdgeStatus status = 4;
}

service GUINotifier {
    rpc SubscribeNode (ClientId) returns (stream NodeMessage);

    rpc SubscribeEdge (ClientId) returns (stream EdgeMessage);

    rpc EndSubscription (ClientId) returns (Empty);
}
